### PR TITLE
[TWINFACES-323] factory pipeline list & factory resource link

### PR DESCRIPTION
**Pull request by pihaizer**
Tasks:
https://alcosi.atlassian.net/browse/TWINFACES-323
https://alcosi.atlassian.net/browse/TWINFACES-296

### CODE REVIEW

File: src/screens/factories/factories.tsx [hunk 1]
```diff
@@ -84,6 +85,7 @@ const colDefs: Record<
 export function Factories() {
   const { searchFactories } = useFactorySearch();
   const { buildFilterFields, mapFiltersToPayload } = useFactoryFilters();
+  const router = useRouter();
```
Comment (kerim-alcosi) [inline src/screens/factories/factories.tsx:88]: add `breadcrumbs` please

` const { setBreadcrumbs } = useBreadcrumbs();`

File: src/screens/factoryPipelines/factoryPipelines.tsx [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+  pipelineStepsCount: {
+    id: "pipelineStepsCount",
+    accessorKey: "pipelineStepsCount",
+    header: "Pipeline Steps",
+  },
+};
+
+export function FactoryPipelines() {
+  const { searchFactoryPipelines } = useFactoryPipelineSearch();
+  const { buildFilterFields, mapFiltersToPayload } =
```
Comment (kerim-alcosi) [inline src/screens/factoryPipelines/factoryPipelines.tsx:135]: add `breadcrumbs` please

` const { setBreadcrumbs } = useBreadcrumbs();`

File: src/screens/factories/factories.tsx [hunk 3]
```diff
@@ -12,6 +12,7 @@ import { UserResourceLink } from "@/entities/user";
 import { PagedResponse } from "@/shared/api";
 import { toast } from "sonner";
 import { formatToTwinfaceDate } from "@/shared/libs";
+import {useRouter} from "next/navigation";
```
Comment (kerim-alcosi) [inline src/screens/factories/factories.tsx:15]: lets enhance columns:
- factoryBranchesCount
- factoryUsagesCount
- factoryErasersCount
- factoryMultipliersCount
- factoryPipelinesCount

via

```tsx
    cell: (data) => (
      <div className="text-center">{data.getValue<string>()}</div>
    ),
```

File: src/screens/factoryPipelines/factoryPipelines.tsx [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+import { Experimental_CrudDataTable } from "@/widgets";
+import { ColumnDef } from "@tanstack/table-core";
+import { Factory } from "@/entities/factory";
+import {
+  TwinClass_DETAILED,
+  TwinClassResourceLink,
+} from "@/entities/twinClass";
+import { Check } from "lucide-react";
+import { TwinClassStatusResourceLink, TwinStatus } from "@/entities/twinStatus";
+import { FactoryResourceLink } from "@/entities/factory/components/resource-link/resource-link";
```
Comment (kerim-alcosi) [inline src/screens/factoryPipelines/factoryPipelines.tsx:20]: `import { FactoryResourceLink } from "@/entities/factory`

File: src/screens/factoryPipelines/factoryPipelines.tsx [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+    cell: (data) => <>{data.getValue() && <Check />}</>,
+  },
+  inputTwinClass: {
+    id: "inputTwinClass",
+    accessorKey: "inputTwinClass",
+    header: "Input Twin Class",
+    cell: ({ row: { original } }) => (
+      <div className="max-w-48 inline-flex">
+        <TwinClassResourceLink
+          data={original.inputTwinClass as TwinClass_DETAILED}
```
Comment (kerim-alcosi) [inline src/screens/factoryPipelines/factoryPipelines.tsx:78]: need to add `hydration` to `searchFactoryPipelines` so we can remove `as TwinClass_DETAILED `

File: src/screens/factoryPipelines/factoryPipelines.tsx [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+  },
+  nextFactory: {
+    id: "nextFactory",
+    accessorKey: "nextFactory",
+    header: "Next Factory",
+    cell: ({ row: { original } }) => (
+      <div className="max-w-48 inline-flex">
+        {original.nextFactory && (
+          <FactoryResourceLink
+            data={original.nextFactory as Factory}
```
Comment (kerim-alcosi) [inline src/screens/factoryPipelines/factoryPipelines.tsx:107]: need to add `hydration` to `searchFactoryPipelines` so we can remove `as Factory`

File: src/screens/factoryPipelines/factoryPipelines.tsx [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+          />
+        </div>
+      ),
+  },
+  nextFactory: {
+    id: "nextFactory",
+    accessorKey: "nextFactory",
+    header: "Next Factory",
+    cell: ({ row: { original } }) => (
+      <div className="max-w-48 inline-flex">
```
Comment (kerim-alcosi) [inline src/screens/factoryPipelines/factoryPipelines.tsx:104]:
```tsx
{original.nextFactory && (<div className="max-w-48 inline-flex">...</div>}
```

File: src/screens/factories/factories.tsx (hunk 3 above)
Comment (pihaizer) [reply src/screens/factories/factories.tsx:15]: We don't have text-centering on any other columns, why add it here?

**Review by kerim-alcosi — APPROVED** [review]

File: src/screens/factoryPipelines/factoryPipelines.tsx (hunk 5 above)
Comment (kerim-alcosi) [reply src/screens/factoryPipelines/factoryPipelines.tsx:78]: https://alcosi.atlassian.net/browse/TWINFACES-419

**Pull request merged**